commit 14fafef176696aa5b4c21fbdc0d8714ddf53bbb3 (refs/changes/24/2024/1)
Author: Marsell Kukuljevic <marsell@joyent.com>
Date:   2017-06-01T23:35:58+12:00 (2 years, 4 months ago)
    
    PUBAPI-1399: fabrics.test.js failing with multi-process testing due to caching

diff --git a/lib/index.js b/lib/index.js
index 7904ddd..0da4102 100644
--- a/lib/index.js
+++ b/lib/index.js
@@ -840,7 +840,7 @@ UFDS.prototype._getUser = function _getUser(filter, msg, account, cb, noCache) {
                         }
                         result.dclocalconfig = data;
                         return callback();
-                    });
+                    }, noCache);
                 }
             ]
         }, function (postErr, res) {
@@ -1748,11 +1748,13 @@ UFDS.prototype.listForeigndc = function listForeigndc(user, authdev, cb) {
  * @param {String} user - user uuid.
  * @param {String} account - (Optional) main account uuid for subusers.
  * @param {Function} cb standard callback.
+ * @param {Boolean} noCache optional flag to force skipping the cache.
  */
 UFDS.prototype.listDcLocalConfig =
-function listDcLocalConfig(uuid, account, cb) {
+function listDcLocalConfig(uuid, account, cb, noCache) {
     var base;
     if (typeof (account) === 'function') {
+        noCache = cb;
         cb = account;
         account = null;
     }
@@ -1787,7 +1789,7 @@ function listDcLocalConfig(uuid, account, cb) {
                 'Multiple dclocalconfig results');
             return cb(_err);
         }
-    });
+    }, noCache);
 };
 
 /**
@@ -1800,11 +1802,13 @@ function listDcLocalConfig(uuid, account, cb) {
  * @param {String} uuid - string uuid of user.
  * @param {String} account - (Optional) main account uuid for subusers.
  * @param {Function} cb standard callback.
+ * @param {Boolean} noCache optional flag to force skipping the cache.
  */
 UFDS.prototype.getDcLocalConfig =
-function getDcLocalConfig(uuid, account, datacenter, cb) {
+function getDcLocalConfig(uuid, account, datacenter, cb, noCache) {
     var dn;
     if (typeof (datacenter) === 'function') {
+        noCache = cb;
         cb = datacenter;
         datacenter = account;
         account = null;
@@ -1830,7 +1834,7 @@ function getDcLocalConfig(uuid, account, datacenter, cb) {
             cfg = data[0];
         }
         return cb(null, cfg);
-    });
+    }, noCache);
 };
 
 /**
@@ -1947,7 +1951,7 @@ function updateDcLocalConfig(uuid, account, datacenter, update, cb) {
                 });
             }
         }
-    });
+    }, true);
 };
 
 /**
